.github/workflows/build.yml

```diff
@@ -18,7 +18,7 @@ jobs:
         - default
         include:
         - os: ubuntu-22.04
-          gcc: 12
+          gcc: 12.1
```

Comment:
This variable gets interpreted as part of a package name, and Ubuntu has no `gcc-12.1` package (only major versions are used). 

https://packages.ubuntu.com/search?keywords=gcc-12
https://packages.ubuntu.com/search?keywords=gcc-12.1